Surface_mesh_simplification/include/CGAL/Surface_mesh_simplification/Policies/Edge_collapse/GarlandHeckbert_plane_and_line_policies.h

```diff
@@ -0,0 +1,127 @@
+  typedef typename internal_np::Lookup_named_param_def<internal_np::vertex_normal_map_t,
+                                                       NamedParameters,
+                                                       Vertex_normal_dmap>::type    Vertex_normal_map;
+
+  Vertex_normal_map vnm(const TriangleMesh& tmesh, const NamedParameters& np) const{
+    using parameters::choose_parameter;
+    using parameters::is_default_parameter;
+    using parameters::get_parameter;
+    constexpr bool must_compute_vertex_normals = is_default_parameter<NamedParameters, internal_np::vertex_normal_map_t>::value;
+
+    Vertex_normal_map vertex_normals = choose_parameter(get_parameter(np, internal_np::vertex_normal_map),
+                                                        get(Vertex_normal_tag(), tmesh));
+    if constexpr(must_compute_vertex_normals){
+      Polygon_mesh_processing::compute_vertex_normals(tmesh, vertex_normals, np);
+    }
+    return vertex_normals;
+  }
+
+  double lw(const NamedParameters& np) const{
+    using parameters::choose_parameter;
+    using parameters::get_parameter;
+    return choose_parameter(get_parameter(np, internal_np::line_policies_weight), 0.01);
+  }
+
+  FT dm(const NamedParameters& np) const {
+    using parameters::choose_parameter;
+    using parameters::get_parameter;
+    return choose_parameter(get_parameter(np, internal_np::discontinuity_multiplier), 100);
+  }
+};
+
+}
+
+template<typename TriangleMesh, typename GeomTraits>
+class GarlandHeckbert_plane_and_line_policies
+  : public internal::GarlandHeckbert_composed_policies<TriangleMesh, GeomTraits,
+                                             GarlandHeckbert_plane_policies<TriangleMesh, GeomTraits>,
+                                             internal::GarlandHeckbert_line_policies<TriangleMesh, GeomTraits>,
+                                             true>
+{
+  typedef GarlandHeckbert_plane_policies<TriangleMesh, GeomTraits>                            GH_plane_polices;
+  typedef internal::GarlandHeckbert_line_policies<TriangleMesh, GeomTraits>                   GH_line_polices;
+  typedef internal::GarlandHeckbert_composed_policies<TriangleMesh, GeomTraits,
+                                                      GH_plane_polices,
+                                                      GH_line_polices,
+                                                      true>                                   Base;
+
+private:
+  using TM=TriangleMesh;
+  using GT=GeomTraits;
+
+
+public:
+  typedef typename Base::Quadric_calculator Quadric_calculator;
+
+  typedef typename GeomTraits::FT                                              FT;
+
+public:
+  template<typename NP = parameters::Default_named_parameters>
+  GarlandHeckbert_plane_and_line_policies(TriangleMesh& tmesh, const NP& np = parameters::default_values()):
+    Base(tmesh,
+      GH_plane_polices(tmesh),
+      GH_line_polices(tmesh, internal::GH_helper<TM,NP>().vnm(tmesh, np)),
```

Comment:
The NP type is really a very specific type that mirrors the actual instance "np", so it's a bit awkward to separate them (one is the template parameter of the type, and the other one is a parameter, later on). I'd either do:
```cpp
internal::GH_helper<TM,NP>(np).dm()) // NPs are lightweight objects so it's fine to copy them
```
or
```cpp
internal::GH_helper::dm(tmesh, np) // static function
```